contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -2577,6 +2577,254 @@ AS 'babelfishpg_tsql', 'update_user_catalog_for_guest';
+  	    ELSE NULL
+  	  END
+  	AS sys.int) AS CHAR_OCTET_LENGTH
+  , CAST(
+  	  CASE
+  	    WHEN ss.x IS NULL THEN 0
+  	    ELSE (ss.x).n 
+  	  END 
+  	AS sys.int) AS ORDINAL_POSITION
+  , CAST(
+      CASE
+        WHEN ss.x IS NULL AND ss.proretset THEN 'NO'
+        WHEN st.is_table_type = 1 THEN 'YES'
+        WHEN sdit.nullable = 1 THEN 'YES'
+        ELSE 'NO'
+      END
+    AS sys.varchar(254)) AS IS_NULLABLE
+  , CAST(
+  	  CASE
+  	    WHEN ss.x IS NULL THEN
+  	      CASE
+  	        WHEN ss.proretset THEN 0
+  	        WHEN ss.prokind = 'p' THEN 56
+  	        ELSE sdit.ss_data_type
+  	      END
+  	    WHEN st.is_table_type = 1 THEN 0
+  	    ELSE sdit.ss_data_type
+  	  END
+  	AS sys.tinyint) AS SS_DATA_TYPE
+  , CAST(ss.proname AS sys.sysname) AS original_procedure_name
+  FROM 
+  ( 
+    -- CTE to query procedures related to bbf
+     WITH bbf_proc AS (
+   	   SELECT
+         p.proname as proname,
+         p.proargnames as proargnames,
+         p.proargmodes as proargmodes,
+         p.prokind as prokind,
+         p.proretset as proretset,
+         p.prorettype as prorettype,
+         p.pronamespace as pronamespace,
+         p.proallargtypes as proallargtypes,
+         p.proargtypes as proargtypes
+       FROM 
+         pg_proc p
+       WHERE (
+	     p.pronamespace in (select schema_id from sys.schemas union all select oid from pg_namespace where nspname = 'sys')
+	     AND (pg_has_role(p.proowner, 'USAGE') OR has_function_privilege(p.oid, 'EXECUTE'))
+	     AND (p.probin like '{%typmod_array%}'-- typmod array exists for those that are user-defined
+	       OR p.proname like 'sp\_%' -- filter out internal babelfish-specific procs
+	       OR p.proname like 'xp\_%'
+	       OR p.proname like 'dm\_%'
+	       OR p.proname like 'fn\_%'))
+	)
+	       
+     SELECT -- Selects all parameters (input and output), but NOT return values
+       p.proname as proname,
+       p.proargnames as proargnames,
+       p.proargmodes as proargmodes,
+       p.prokind as prokind,
+       p.proretset as proretset,
+       p.prorettype as prorettype,
+       p.pronamespace as pronamespace,
+       information_schema._pg_expandarray(
+       COALESCE(p.proallargtypes,
+         CASE 
+           WHEN p.prokind = 'f' THEN (CAST(p.proargtypes AS oid[]))
+           ELSE CAST(p.proargtypes AS oid[])
+         END
+       )) AS x
+    FROM bbf_proc p
+      
+    UNION ALL
+      
+    SELECT -- Selects all return values
+      p.proname as proname,
+      p.proargnames as proargnames,
+      p.proargmodes as proargmodes,
+      p.prokind as prokind,
+      p.proretset as proretset,
+      p.prorettype as prorettype,
+      p.pronamespace as pronamespace,
+      NULL AS x -- null value indicates that we are retrieving the return values of the proc/func
+    FROM bbf_proc p
+  ) ss
+  INNER JOIN pg_catalog.pg_namespace ns ON ns.oid = ss.pronamespace
+  LEFT JOIN sys.babelfish_namespace_ext bne ON ns.nspname = bne.nspname 
```

Comment:
Unless we want to show objects out side of BBF, I think LEFT JOIN in 2017/2018 could be changed into INNER JOIN